src/server.cpp

```diff
@@ -2151,6 +2151,20 @@ void Server::sendAddNode(v3s16 p, MapNode n, u16 ignore_id,
 	}
 }
 
+void Server::sendCurrentNode(session_t peer_id, v3s16 p, MapNode n)
+{
+	if (m_env->getMap().getNodeMetadata(p)) {
```

Comment:
Does not check whether it actually contains metadata - only whether the container exists (which is always the case).

Reply:
I firstly used `m_env->getMap().getNodeMetadata(p).empty()`, but I got memory access errors, so I noticed that the function returns meta or NULL:
https://github.com/minetest/minetest/blob/3a95d760e97d16fcc499aee8df1be4fe583c8168/src/nodemetadata.cpp#L203-L209
https://github.com/minetest/minetest/blob/3a95d760e97d16fcc499aee8df1be4fe583c8168/src/map.cpp#L963-L980 

When testing with a chest it worked correctly.